src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/active_directory_service_settings.rb

```diff
@@ -19,20 +19,44 @@ class ActiveDirectoryServiceSettings
     #
     # @return [ActiveDirectoryServiceSettings] settings required for authentication.
     def self.get_azure_settings
-      settings = ActiveDirectoryServiceSettings.new
-      settings.authentication_endpoint = 'https://login.windows.net/'
```

Comment:
Yes https://login.microsoftonline.com/ one is new endpoint that we should use. The one that we were using (https://login.windows.net/) is the old endpoint and as per my conversation with @amarzavery it's for backward compatibility.